.github/workflows/sanity-test.yml

```
@@ -95,6 +95,10 @@ jobs:
         run: |
           ./deploy-test-resources.sh
 
+      - name: Test Proxy
+        run: |
+          go test ./test/proxy-tests/
```

I wonder if we should make it more general.
It will allow us to add more tests in the future without having to go through this step again.

```suggestion
      - name: Run go tests
        run: |
          go test ./test/go-tests/
```